packages/fuels-contract/src/contract.rs

```diff
@@ -550,14 +581,24 @@ where
         let attempts = max_attempts.unwrap_or(DEFAULT_TX_DEP_ESTIMATION_ATTEMPTS);
 
         for _ in 0..attempts {
-            let result = self.call_or_simulate(true).await;
+            let result = self.simulate_without_decode().await;
 
             match result {
                 Err(Error::RevertTransactionError(_, receipts))
                     if ContractCall::is_missing_output_variables(&receipts) =>
                 {
                     self = self.append_variable_outputs(1);
                 }
+
+                Err(Error::RevertTransactionError(_, ref receipts)) => {
+                    if let Some(receipt) = ContractCall::find_contract_not_in_inputs(receipts) {
+                        let contract_id = Bech32ContractId::from(*receipt.contract_id().unwrap());
+                        self = self.append_contract(contract_id);
+                    } else {
+                        return Err(result.err().unwrap());
```

Comment:
This error here will be _very_ cryptic when returned to the user. 

The user will be trying to `estimate_tx_dependencies`, but if it _is_ an `Error::RevertTransactionError` and for some weird reason that's out of our control, `ContractCall::find_contract_not_in_inputs` returns `None`, the `Err` here will be _very_ hard to understand.

So we should return this `Err` with a string saying something like `"Couldn't estimate tx dependencies because we couldn't find the missing contract input"`.

Reply:
If `ContractCall::find_contract_not_in_inputs` returns `None` it should mean that the `Error::RevertTransactionError` was due to reasons other than missing output variables or contract inputs. We should forward the original error to the user in that case since it cannot be fixed by estimating dependencies anyways.

Reply:
This was the initial idea.